### ISSUE TITLE
[MU4 Issue] 4 measure repeat doesn't wrap correctly, hangs off staves

### ISSUE THREAD

**Issue by hackbar** [musescore/MuseScore#11571]
**Describe the bug**
4 measure repeat doesn't wrap correctly, hangs off staffs

**To Reproduce**
Not sure, I think just enter a four measure repeat and get a wrap in the middle of it.

Here's the file I with the bug: https://drive.google.com/file/d/1fQE0NHeqvS-JsNCd2i9rAWDC7qio_VyR/view?usp=sharing

**Expected behavior**
I think the four bars should only wrap at the beginning or end, not in the middle.

**Screenshots**
https://drive.google.com/file/d/1biuVd74m9sqcV_TLOSHdttnCqgX7xPsP/view?usp=sharing

**Platform information**
 - OS: Linux

**Comment by cbjeukendrup** [comment]
Normally, a measure repeat comes together with "group measure" elements; that's a special element that can be added to a barline, and keeps the measures before and after that barline on the same system. If these elements were not created for you, or you accidentally deleted them, you can find them in the Palettes and (re)apply them to the barlines manually, so that it will look like this:
<img width="807" alt="Schermafbeelding 2022-05-09 om 00 04 14" src="https://user-images.githubusercontent.com/48658420/167317616-8b60d194-79a4-48a2-abe3-f7f7a878613d.png">

**Comment by hackbar** [comment]
Those symbols seem to still be present in the score, yet this happens.

I've noticed some weird behaviors with page size I'm still trying to reproduce, but if you don't see it in the score I uploaded, trying tweaking the page size. Just get it so the second measure of the four measure repeat is the last one a line, and you'll see it.

**Comment by Tantacrul** [comment]
I've downloaded the score but am not understanding the issue. Could you show:

1. A visual of what is wrong?
2. A visual of expected behaviour?

Thanks!

**Comment by cbjeukendrup** [comment]
@Tantacrul the problem occurs in the Soprano Saxophone part. The four selected measures should be grouped together:
<img width="658" alt="Schermafbeelding 2022-05-14 om 23 50 39" src="https://user-images.githubusercontent.com/48658420/168449163-ac61014c-d562-4ea8-8563-a5ee99cdb2dd.png">

Normally, when creating a measure repeat, those "Group measure" elements are added, as I showed before:
<img width="807" alt="image" src="https://user-images.githubusercontent.com/48658420/168468656-fc7f8cf3-36d7-481a-bf1d-61bffa9899e0.png">

But it turns out that these are only created in the part (or score) where you are currently working, and not in any other parts! That leads to this situation of potential measure breaks in the middle of a measure repeat.

So that is what's really going on here. It may be a bit tricky to solve this in a good way, because obviously not _all_ parts will show the staff that contains the measure repeat, and in the parts that don't, the "Group measures" elements don't make sense. And what if the user _removes_ the staff containing the measure repeat, should we be so smart that we remove the "Group measures" elements too?

A more elegant solution would be not to add separate "Group measures" elements when adding a measure repeat, but to "infer" the measure groupings from the measure repeat itself. If we do that, we wouldn't have to worry about all of the above. I don't know why it wasn't done that way initially, maybe because it was difficult. But I'm sure @mike-spa would be able to find a solution :)

**Comment by MarcSabatella** [comment]
As I recall, the reason the grouping was done explicitly was just that this code already existed, since the “no break” element was already implemented early in the 3.0 development cycle.  It was just never finished and thus was was disabled before release.  I agree that it’s probably simpler to just infer it as needed here.

**Comment by Tantacrul** [comment]
Yeah, this one found its way in under the radar and doesn't seem to have been tested extensively. It's a pity putting something like this in, which no one thinks is the right solution, which we'll then have to live with, solve bugs for and ultimately replace in a messy way later.

**Comment by cbjeukendrup** [comment]
Closing as superseded by https://github.com/musescore/MuseScore/issues/26836
